### PR TITLE
DOC: update documentation of type annotation (PEP 585)

### PR DESCRIPTION
**Pull request by twoertwein**

### CODE REVIEW

File: doc/source/development/contributing_codebase.rst [hunk 1]
```diff
@@ -315,21 +315,31 @@ You should write
...
 
-   primes: List[int] = []
+``Optional`` should be  avoided in favor of the shorter ``| None``, so instead of
 
-``Optional`` should be used where applicable, so instead of
+.. code-block:: python
+
+   from typing import Union
+
+   maybe_primes: list[Union[int, None]] = []
```
Comment (jreback) [inline doc/source/development/contributing_codebase.rst:326]: can use `|` ?

**Review by jreback — CHANGES_REQUESTED** [review]
small comment

**Comment by jreback** [comment]
pre commit failing

File: doc/source/development/contributing_codebase.rst (hunk 1 above)
Comment (twoertwein) [reply doc/source/development/contributing_codebase.rst:326]: These are two negative examples followed by one positive example.

**Comment by twoertwein** [comment]
Without the CSS:
image

**Review by jreback — APPROVED** [review]

**Pull request merged**